pkg/resources/sts/sts.go

```diff
@@ -0,0 +1,70 @@
+package sts
+
+import (
+	"context"
+	"fmt"
+	integreatlyv1alpha1 "github.com/integr8ly/integreatly-operator/apis/v1alpha1"
+	"github.com/integr8ly/integreatly-operator/pkg/addon"
+	"github.com/integr8ly/integreatly-operator/pkg/resources/logger"
+	cloudcredentialv1 "github.com/openshift/api/operator/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	clusterCloudCredentialName  = "cluster"
+	RoleArnParameterName        = "sts-role-arn"
+	RoleSessionName             = "Red-Hat-cloud-resources-operator"
+	CredsSecretName             = "sts-credentials"
+	CredsSecretRoleARNKeyName   = "role_arn"
+	CredsSecretTokenPathKeyName = "web_identity_token_file"
+)
+
+func IsClusterSTS(ctx context.Context, client k8sclient.Client, log logger.Logger) (bool, error) {
+	cloudCredential := &cloudcredentialv1.CloudCredential{}
+	if err := client.Get(ctx, k8sclient.ObjectKey{Name: clusterCloudCredentialName}, cloudCredential); err != nil {
+		log.Error("failed to get cloudCredential whle checking if STS mode", err)
+		return false, err
+	}
+
+	if cloudCredential.Spec.CredentialsMode == cloudcredentialv1.CloudCredentialsModeManual {
```

Comment:
Is there any chance that the ManualMode could be used for other cases aside from STS? Would it be safer to get the secret and see that it has an arn in it? I n the case of non sts I believe there will be no arn in the secret.

Reply:
Seems to be used for when the operator is in a disabled mode 
* https://github.com/openshift/cloud-credential-operator/search?p=1&q=CloudCredentialsModeManual&type=code 
* https://github.com/openshift/cloud-credential-operator/blob/master/docs/mode-manual-creds.md

Not sure, is there a way to detect that the cluster is running in an AWS STS configuration :shrug: 

But yeah, probably be better to check for an ARN in the secret :thinking:

Reply:
I came to the same conclusion from those docs. It's likely this is fine as it is and either method would give the same result.

Reply:
@laurafitzgerald Should I update this to use the secret or you good to keep it as is ? As you said I think either or would work :thinking: